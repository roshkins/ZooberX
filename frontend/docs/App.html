<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: App</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: App</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>App<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">The full ZooberX App, including state and UI.</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="App"><span class="type-signature"></span>new App<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Sets up state with sample data and destination.
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line10">line 10</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="componentDidMount"><span class="type-signature"></span>componentDidMount<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    After component mounts start running our fetch requests to update state
  so that state can be set properly.
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line89">line 89</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="getNearestBeest"><span class="type-signature"></span>getNearestBeest<span class="signature">(latitude, longitude, direction)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    runs getYourWildebeest on the server and stores the result in the state.
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>latitude</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The current latitude of the Oxpecker.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>longitude</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The current longitude of the Oxpecker.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>direction</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The current direction of the Oxpecker.</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line110">line 110</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Renders the user interface.
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line128">line 128</a>
    </li></ul></dd>
    

    

    

    
</dl>













<h5>Returns:</h5>

        
<div class="param-desc">
    A React element that represents the user interface.
</div>



    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="App.html">App</a></li><li><a href="Form.html">Form</a></li><li><a href="Map.html">Map</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.4</a> on Sun Sep 03 2017 07:07:08 GMT-0700 (Pacific Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>