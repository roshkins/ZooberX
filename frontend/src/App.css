@import url('https://fonts.googleapis.com/css?family=Open+Sans+Condensed:300,300i,700');

.Map {
  flex: 2;
  height: 100%;
}
.App {
  display: flex;
  width: 100%;
  height: 100%;
  font-family: 'Open Sans Condensed', sans-serif;
}

select,
input {
  font-family: 'Open Sans Condensed', sans-serif;
  font-size: 18pt;
}

.Form {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 100%;
  padding: 2em;
  font-size: 18pt;
}
.LeftColumn {
  flex: 1;
  display: flex;
  flex-direction: column;
}

.Logo {
  font-family: 'Open Sans Condensed', sans-serif;
  font-weight: normal;
  font-size: 72pt;
  color: #fff;
  background-color: #000;
  padding: 0.1em;
}
.x {
  font-weight: bold;
}

.Submit {
  background-color: #000;
  color: #fff;
  border: none;
  padding: 0.1em;
  cursor: pointer;
}
