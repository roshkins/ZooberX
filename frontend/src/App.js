import React, { Component } from "react";
import "./App.css";
import Map from "./Map";

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="logo">ZooberX</div>
        <div className="locationForm">
          <form>
            <label htmlFor="latitude">Your latitude</label>
            <input type="text" name="latitude" />
            <label htmlFor="longitude">Your longitude</label>
            <input type="text" name="longitude" />
            <label htmlFor="direction">What is your destination?</label>
            <select name="direction">
              <option value="Kenya">Kenya</option>
              <option value="Tanzania">Tanzania</option>
            </select>
            <input type="submit" value="Request" />
          </form>
        </div>
        <div className="map">
          <Map
            containerElement={<div style={{ height: `100%` }} />}
            mapElement={<div style={{ height: `100%` }} />}
          />
        </div>
      </div>
    );
  }
}

export default App;
