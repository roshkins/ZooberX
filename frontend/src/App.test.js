import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import renderer from "react-test-renderer";

/** make sure the app can survive rendering */
it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
