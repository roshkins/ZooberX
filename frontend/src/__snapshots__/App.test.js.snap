// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="logo"
  >
    ZooberX
  </div>
  <div
    className="locationForm"
  >
    <form>
      <label
        htmlFor="latitude"
      >
        Your latitude
      </label>
      <input
        name="latitude"
        type="text"
      />
      <label
        htmlFor="longitude"
      >
        Your longitude
      </label>
      <input
        name="longitude"
        type="text"
      />
      <label
        htmlFor="direction"
      >
        What is your destination?
      </label>
      <select
        name="direction"
      >
        <option
          value="Kenya"
        >
          Kenya
        </option>
        <option
          value="Tanzania"
        >
          Tanzania
        </option>
      </select>
      <input
        type="submit"
        value="Request"
      />
    </form>
  </div>
  <div
    className="map"
  >
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
          }
        }
      />
      <div />
    </div>
  </div>
</div>
`;
